src/plugins/dashboard/public/dashboard_app/top_nav/dashboard_top_nav.tsx

```diff
@@ -160,8 +154,25 @@ export function DashboardTopNav({ embedSettings, redirectTo }: DashboardTopNavPr
               }
             : undefined,
       },
+    ];
+
+    setBreadcrumbs([
+      {
+        text: getDashboardBreadcrumb(),
+        'data-test-subj': 'dashboardListingBreadcrumb',
+        onClick: () => {
+          redirectTo({ destination: 'listing' });
+        },
+      },
+      ...dashboardTitleBreadcrumbs,
     ]);
-  }, [setBreadcrumbs, redirectTo, dashboardTitle, dashboard, viewMode]);
+
+    if (serverless?.setBreadcrumbs) {
+      // set serverless breadcrumbs if available,
+      // set only the dashboardTitleBreadcrumbs because the main breadcrumbs automatically comes as part of the navigation config
+      serverless.setBreadcrumbs(dashboardTitleBreadcrumbs);
+    }
```

Comment:
should we have an `else` branch here to make it clearer that it's one or the other?